library/src/tensile_host.cpp

```diff
@@ -334,8 +337,9 @@ namespace
             std::string path;
             path.reserve(PATH_MAX);
 
-            auto        pAMDGPU   = std::dynamic_pointer_cast<Tensile::AMDGPU>(hardware);
-            std::string processor = GetProcessorName(pAMDGPU->processor);
+            // The name of the current GPU platform
+            static const std::string processor
+                = "gfx" + std::to_string(_rocblas_handle::device_arch_id());
```

Comment:
We use `device_arch_id` in other places, so this was simplifying it to avoid having to list all the different cases. If there's a more standard way of returning the device ID as a string, we should change all places in rocBLAS to use it.

Reply:
[Other ROCm code](https://github.com/ROCm-Developer-Tools/HIP/search?q=gcnArch&unscoped_q=gcnArch) uses the `std::to_string( int_arch_id )` method, but only if it's a Graphics Core Next. Another PR should probably refactor rocBLAS to return a string (remembering to cache it in a local static variable). Right now an integer is returned because we compare it to `906` in a few places.